docs/reference/server-events.md

```diff
@@ -0,0 +1,982 @@
+  </template>
+</AposCodeBlock>
+
+## `@apostrophecms/login`
+
+### `deserialize`
+
+Triggered after a user logs in and their data is retrieved.
+
+#### Parameters
+
+- `user`: The user's data from the database
+
+<AposCodeBlock>
+  ```javascript
+  handlers(self, options) {
+    return {
+      'deserialize': {
+        async handlerName(user) { ... }
+      }
+    };
+  }
+  ```
+  <template v-slot:caption>
+    modules/@apostrophecms/login/index.js
+  </template>
+</AposCodeBlock>
+
+## `@apostrophecms/migration`
+
+### `after`
+
+Triggered after all data migrations have run. The database is now in a stable state.
+
+There is no data included with the event for handlers.
+
+<AposCodeBlock>
+  ```javascript
+  handlers(self, options) {
+    return {
+      'after': {
+        async handlerName() { ... }
+      }
+    };
+  }
+  ```
+  <template v-slot:caption>
+    modules/@apostrophecms/migration/index.js
+  </template>
+</AposCodeBlock>
+
+## `@apostrophecms/page`
+
+### `beforeUnpublish`
+
+Triggered just before a page document is unpublished (converted to a draft document).
+
+#### Parameters
+
+- `req`: The active request
+- `published`: The published document that is about to be unpublished
+
+<AposCodeBlock>
+  ```javascript
+  handlers(self, options) {
+    return {
+      'beforeUnpublish': {
+        async handlerName(req, published) { ... }
+      }
+    };
+  }
+  ```
+  <template v-slot:caption>
+    modules/@apostrophecms/page/index.js
+  </template>
+</AposCodeBlock>
+
+### `afterParkAll`
+
+Triggered after [parked pages](/reference/module-api/module-options.md#park) have been saved or updated in the database (as appropriate) during app startup.
+
+There is no data included with the event for handlers.
+
+<AposCodeBlock>
+  ```javascript
+  handlers(self, options) {
+    return {
+      'afterParkAll': {
+        async handlerName() { ... }
+      }
+    };
+  }
+  ```
+  <template v-slot:caption>
+    modules/@apostrophecms/page/index.js
+  </template>
+</AposCodeBlock>
+
+### `afterConvert`
+
+Triggered after page data is run through the schema module's `convert` method, validating its data to the page type's field schema.
+
+#### Parameters
+
+- `req`: The active request
+- `data`: The data being saved to the page, prior to schema conversion
+- `page`: The page as it will be saved, following schema conversion
+
+<AposCodeBlock>
+  ```javascript
+  handlers(self, options) {
+    return {
+      'afterConvert': {
+        async handlerName(req, data, page) { ... }
+      }
+    };
+  }
+  ```
+  <template v-slot:caption>
+    modules/@apostrophecms/page/index.js
+  </template>
+</AposCodeBlock>
+
+### `beforeMove`
+
+Triggered just *before* a page is moved from one position in the page tree to another.
+
+#### Parameters
+
+- `req`: The active request
+- `page`: The page being moved
+- `target`: The page used as a target for positioning in the page tree
+- `position`: The position to place the page in relation to the target
```

Comment:
should reference the `move` docs or spell out the possible values (it'll be one of the named values, any numeric position has already been resolved to a named value at this point)